synapse/handlers/room_batch.py

```diff
@@ -0,0 +1,373 @@
+import logging
+from typing import TYPE_CHECKING, List, Tuple
+
+from synapse.http.servlet import (
+    assert_params_in_dict,
+)
+from synapse.api.constants import EventContentFields, EventTypes
+from synapse.appservice import ApplicationService
+from synapse.types import JsonDict, Requester, UserID, create_requester
+from synapse.util.stringutils import random_string
+
+if TYPE_CHECKING:
+    from synapse.server import HomeServer
+
+logger = logging.getLogger(__name__)
+
+
+class RoomBatchHandler:
```

Comment:
Functionality is the same. Just copying logic from `synapse/rest/client/room_batch.py`

([Complement tests](https://github.com/matrix-org/complement/blob/0e57c33813c8da77004b539bfff2fe320410b612/tests/msc2716_test.go) pass)